operator/pkg/apis/validation/validation.go

```diff
@@ -125,6 +125,14 @@ func detectCniIncompatibility(client kube.Client, cniEnabled bool, ztunnelEnable
 			errs = util.AppendErr(errs,
 				errors.New("detected Cilium CNI with 'bpf-lb-sock=true'; this requires 'bpf-lb-sock-hostns-only=true' to be set"))
 		}
+		kpr := cilium.Data["kube-proxy-replacement"]
+		if kpr == "strict" {
```

Comment:
`kube-proxy-replacement: strict` is used only in older Cilium versions so seems like this may not be quite correct in all cases at least. Now its just true/false IIUC

Reply:
Ah but it seems like maybe only the old ciliums have this behavior in the first place... pending some research

Reply:
```
Cilium 0.15: --set kubeProxyReplacement=strict 
  kube-proxy-replacement: "strict"
  bpf-lb-sock: "false"
Behavior: broken
Add --set socketLB.hostNamespaceOnly=true
  kube-proxy-replacement: "strict"
  bpf-lb-sock: "false"
  bpf-lb-sock-hostns-only: "true"
Behavior: works

Cilium 0.16: can no longer set 'strict'.
With --set kubeProxyReplacement=true
  kube-proxy-replacement: "true"
  bpf-lb-sock: "false"
  bpf-lb-sock-hostns-only: "true"
Behavior: works
```

OK so I think this is correct. It would be good to put a comment of something along the lines of "Older versions of cilium used 'strict' and defaulted to hostns-only=false; newer version use true/false and default to hostns-only=true"